CONTRIBUTING.md

```diff
@@ -0,0 +1,284 @@
+  * [Error message guidelines](#error-message-guidelines)
+  * [Test guidelines](#test-guidelines)
+  * [Comment guidelines](#comment-guidelines)
+  * [Running unit tests](#running-unit-tests)
+  * [Running integration tests](#running-integration-tests)
+  * [Adding new dependencies to the project](#adding-new-dependencies-to-the-project)
+* [Releasing](#releasing)
+
+## How to start a contribution
+
+The team always encourages early communication for all types of contributions. Found a bug or see something that could be improved? Open an issue. Want to address something bigger like adding new package manager support or overhauling the entire project? Open an issue or start a Discussion on Github. This way, we can give you guidance and avoid your work being wasted on an implementation which does not fit the project's scope and goal.
+
+Alternatively, submit a pull request with one of the following
+
+* A high-level design of the feature, highlighting goals and key decision points.
+* A proof-of-concept implementation.
+* In case the change is trivial, you can start with a draft or even provide a PR with the final implementation.
+
+### How we deal with larger features
+
+Implementing a larger feature (such as adding a new package manager) is usually a very long and detailed effort. This type of work does not fit well into a single pull request; after several comment threads it becomes almost unmanageable (for you) and very hard to review (for us). For that reason, we request that larger features be split into a series of pull requests. Once approved, these pull requests will be merged into "main", but the new feature will be marked as experimental, and will retain this mark until it meets code quality standards and all necessary changes are merged.
+
+This has several implications
+
+* Experimental features are not fully endorsed by the maintainers, and maintainers will not provide support.
+* Experimental features are not production-ready and should never be used in production.
+* Always expect that an experimental feature can be fully dropped from this project without any prior notice.
+* A feature toggle is needed to allow users to opt-in. This is currently being handled by the `dev-package-managers` flag.
+* All SBOMs produced when an experimental feature is used will be marked as such.
+
+If, for some reason, you feel this proposed workflow does not fit the feature you're contributing, please reach out to the maintainers so we can provide an alternative.
+
+#### Making experimental features production-ready
+
+When a feature's development has reached a stable point, you can propose making it an official part of the project. This signals to users that the feature is production-ready. To communicate this intent to the maintainers, open a pull request containing an Architecture Decision Record (ADR) with an outline of the implementation, and a clear statement of all decisions which were made (as well as their rationale).
```

Comment:
This is more than just being ready. It also signals that maintainers are confident in the feature and have enough understanding and means to maintain it once implementation is finished. I suggest adding
```
Once maintainers are confident that they have enough information to maintain the new feature as officially supported they will accept it and help with moving out from under experimental flag.```

Reply:
Done

Reply:
:taco: